Assets/Mirror/Runtime/NetworkConnection.cs

```diff
@@ -237,30 +237,44 @@ public virtual void TransportReceive(byte[] buffer)
             //
             // let's catch them all and then disconnect that connection to avoid
             // further attacks.
-            try
+            // unpack message
+            NetworkReader reader = new NetworkReader(buffer);
+            if (MessagePacker.UnpackMessage(reader, out int messageId))
             {
-                // unpack message
-                NetworkReader reader = new NetworkReader(buffer);
-                if (MessagePacker.UnpackMessage(reader, out int msgType))
+                if (logNetworkMessages)
                 {
-                    if (logNetworkMessages)
-                    {
-                        Debug.Log("ConnectionRecv con:" + connectionId + " msgType:" + msgType + " content:" + BitConverter.ToString(buffer));
-                    }
-
-                    // try to invoke the handler for that message
-                    if (InvokeHandler(msgType, reader))
+                    Debug.Log($"TransportReceive connection: {connectionId} MessageID: {messageId} Content: {BitConverter.ToString(buffer)}");
+                }
+                try
+                {
+                    if (InvokeHandler(messageId, reader))
                     {
                         lastMessageTime = Time.time;
                     }
                 }
-                else Debug.LogError("HandleBytes UnpackMessage failed for: " + BitConverter.ToString(buffer));
-            }
-            catch (Exception exception)
-            {
-                Disconnect();
-                Debug.LogWarning("Closed connection: " + connectionId + ". This can happen if the other side accidentally (or an attacker intentionally) sent invalid data. Reason: " + exception);
+                // These can happen if the other side accidentally (or an attacker intentionally) sends invalid data
+                catch (OverflowException exception)
+                {
+                    Disconnect(); // Disconnect, this is probably intentional
+                    Debug.LogWarning($"Closed connection: {connectionId}. An invalid integer was encountered while reading.\nMessageID: {messageId}\nContent: {BitConverter.ToString(buffer)}\nException: {exception}");
```

Comment:
why an invalid integer? not float etc.?

Reply:
Only integers overflow in the network reader.